clients/src/main/java/org/apache/kafka/clients/consumer/internals/CompletedFetch.java

```diff
@@ -326,7 +327,17 @@ <K, V> ConsumerRecord<K, V> parseRecord(Deserializers<K, V> deserializers,
                     key, value, headers, leaderEpoch);
         } catch (RuntimeException e) {
             log.error("Deserializers with error: {}", deserializers);
-            throw new RecordDeserializationException(partition, record.offset(),
+            ByteBuffer keyBytes = record.key();
+            byte[] key = org.apache.kafka.common.utils.Utils.toNullableArray(keyBytes);
+            ByteBuffer valueBytes = record.value();
+            byte[] value = Utils.toNullableArray(valueBytes);
+            Headers headers = new RecordHeaders(record.headers());
+            ConsumerRecord<byte[], byte[]> consumerRecord = new ConsumerRecord<>(partition.topic(), partition.partition(), record.offset(),
+                    record.timestamp(), timestampType,
+                    keyBytes == null ? ConsumerRecord.NULL_SIZE : keyBytes.remaining(),
+                    valueBytes == null ? ConsumerRecord.NULL_SIZE : valueBytes.remaining(),
+                    key, value, headers, Optional.empty());
```

Comment:
Any reason to omit the epoch value? Can we do the same as the happy path?

```suggestion
            ConsumerRecord<byte[], byte[]> consumerRecord = new ConsumerRecord<>(partition.topic(), partition.partition(), record.offset(),
                    record.timestamp(), timestampType,
                    keyBytes == null ? ConsumerRecord.NULL_SIZE : keyBytes.remaining(),
                    valueBytes == null ? ConsumerRecord.NULL_SIZE : valueBytes.remaining(),
                    key, value, headers, leaderEpoch);
```